server/src/main/java/org/opensearch/wlm/cancellation/TaskSelectionStrategy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskCancellation;
+
+import java.util.List;
+
+/**
+ * Interface for strategies to select tasks for cancellation.
+ * Implementations of this interface define how tasks are selected for cancellation based on resource usage.
+ */
+public interface TaskSelectionStrategy {
```

Comment:
Given that we will always have only one implementation of this we should remove the interface and keep the abstract class only.

Reply:
Ankit had the same feedback, so i replied to that comment here - https://github.com/opensearch-project/OpenSearch/pull/15151#discussion_r1717523136 and resolving this